### PR TITLE
Add yaml tests for attaching disks

### PR DESCRIPTION
**Pull request by iyovcheva**

### CODE REVIEW

File: blockstore/src/test/yaml/def-suff-b/catalog.tests.bom [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        - type: invoke-effector
+          name: "3. Invoke addExtraHdd effector"
+          effector: addExtraHdd
+
+        - type: test-ssh
+          name: "4. addExtraHdd effector executed successfully"
+          brooklyn.config:
+            command: df -h | grep /mount/brooklyn/d
+            assert.out:
+              contains: /mount/brooklyn/d
```
Comment (m4rkmckenna) [inline blockstore/src/test/yaml/def-suff-b/catalog.tests.bom]: Could you add a test to check if the mounted disk is writable

File: blockstore/src/test/yaml/def-suff-b/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
+
+   TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
```
Comment (bostko) [inline blockstore/src/test/yaml/def-suff-b/LICENSE]: @aledsage @murdoaird @nakomis @iyovcheva Isn't it better if we use Cloudsoft Developer License?

File: blockstore/src/test/yaml/def-suff-h/LICENSE [hunk 3]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
+
```
Comment (bostko) [inline blockstore/src/test/yaml/def-suff-h/LICENSE]: @iyovcheva why do you put separate licenses for def-stuff-*?

File: blockstore/src/test/yaml/def-suff-h/catalog.bom [hunk 4]
```diff
@@ -0,0 +1,59 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (bostko) [inline blockstore/src/test/yaml/def-suff-h/catalog.bom]: Shouldn't be Cloudsoft Developer License?

File: blockstore/src/test/yaml/def-suff-b/reference.json [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+      "name": "download.url",
+      "links": {},
+      "description": "URL pattern for downloading the installer (will substitute things like ${version} automatically)"
+    }, {
+      "type": "java.lang.String",
+      "name": "expandedinstall.dir",
+      "links": {},
+      "description": "Directory for installed artifacts (e.g. expanded dir after unpacking .tgz)"
+    }, {
+      "type": "java.lang.String",
```
Comment (bostko) [inline blockstore/src/test/yaml/def-suff-b/reference.json]: What is the purpose of listing those sensors here?

File: catalog.tests.bom [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+        brooklyn.children:
+
+        - type: assert-up-and-running-initial
+          name: "1. Node up and running"
+
+        - type: test-ssh
+          name: "2. Additional mount devices successfully added"
+          brooklyn.config:
+            command: df -h | grep /mount/brooklyn/h
```
Comment (m4rkmckenna) [inline catalog.tests.bom]: Invalid YAML ... duplicate keys

I think you should combine the multiple commands into 
```YAML
type: test-ssh
  name: "2. Additional mount devices successfully added"
  brooklyn.config:
    command: |
      set -e
      df -h | grep /mount/brooklyn/h
      check 2
      check 3
```

File: catalog.bom [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+brooklyn.catalog:
+  items:
+
```
Comment (bostko) [inline catalog.bom]: I prefer it without blank line.

File: catalog.bom [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    license: Apache-2.0
+    publish:
+      license_code: Apache-2.0
+      reference: reference.json
+    item:
+      services:
+      - type: org.apache.brooklyn.entity.software.base.EmptySoftwareProcess
+        id: espentity
+        brooklyn.initializers:
+          - type: io.brooklyn.blockstore.brooklyn-blockstore:brooklyn.location.blockstore.effectors.ExtraHddBodyEffector
```
Comment (bostko) [inline catalog.bom]: Can we have the catalog item without an effector?
That way it seems they are coupled.
What do you thing about 
- For each cloud Catalog items with initializer.
- For each cloud Catalog items with an effector
- One or (For each cloud) mixed catalog item and effector

File: catalog.tests.bom [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+            - contains: /mount/brooklyn/b
+            - contains: /mount/brooklyn/c
+            - contains: rw
+
+        - type: invoke-effector
+          target: $brooklyn:component("espentity")
+          name: "3. Invoke addExtraHdd effector"
+          effector: addExtraHdd
+          params:
+            location.customizer.fields:
```
Comment (bostko) [inline catalog.tests.bom]: @iyovcheva please update that field to mach what is going to be merged in master.

**Comment by bostko** [comment]
I rebased PR.

**Comment by neykov** [comment]
Superseded by #28.

**Pull request closed without merging**